easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1780,6 +1780,10 @@ def get_ref(self, key):
     def __setitem__(self, key, value):
         """Set value of specified easyconfig parameter (help text & co is left untouched)"""
         if key in self._config:
+            # raise an error if we are replacing a value with templates by a value with templates resolved
+            if "%(" in str(self._config[key][0]) and "%(" not in str(value):
```

Comment:
Maybe do this only when (both?) are already strings as it doesn't make sense IMO otherwise and just slows EB down (even more)
BTW: Outdated comment. Can be removed as the code (check and warning) already says what's done.

Reply:
I can change the comment, but I disagree that it does not make sense. Such erroneous replacements will definitely happen in config opts that are arrays (and not strings).

Reply:
>happen in config opts that are arrays (and not strings).

Ah, true. Good point. I was thinking about the costly conversion and string search when setting int values etc. But I guess most are really strings or contain strings.

Reply:
comment has been adjusted